client/src/components/PrivateRoute.js

```diff
@@ -0,0 +1,65 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { UserContext } from '../globals/UserContext';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Grid from '@material-ui/core/Grid';
+import { makeStyles } from '@material-ui/core/styles';
+
+
+const useStyles = makeStyles((theme) => ({
+    loadingContainer: {
+        paddingTop: 300
+    }
+}))
+
+const PrivateRoute = ({ component: Component, ...rest }) => {
+    const classes = useStyles();
+    const user = useContext(UserContext);
+    const [login, setLogin] = useState(null);
+
+    useEffect(() => {
+        let status;
+        fetch("/getcookie", {
+            method: "GET",
+            headers: {
+                "Content-Type": "application/json"
+            },
+            credentials: "include"
+        })
+            .then(res => {
+                status = res.status;
+                if (status >= 400) throw Error('Fail to fetch data')
+                else return res.json()
+            })
+            .then(res => {
+                if (status === 200 && res.success === true) return setLogin(true);
+                throw Error("error");
+            })
+            .catch(err => {
+                user.setUserId("");
+                return setLogin(false)
+            });
+    }, [])
+
+    const Loading = () => {
+        return (
+            <Grid container direction="row" justify="center" alignItem="center"
+                className={classes.loadingContainer}
+            >
+                <CircularProgress disableShrink />
+            </Grid>
+        )
+    }
+
+    if (login === null) return (
+        <Route {...rest} component={Loading} />
+    );
+    return (
+        <Route {...rest} render={props => (
+            login ? <Component {...props} /> : <Redirect to="/login" />
+        )} />
+
+    );
```

Comment:
@sina-jamshidi @fangyu3 Since I've updated local git from origin dev, the `Files Changed` had been messed up here in Github. 

So this part is where I highlighted the actual change and also the follows, please check if there are any problems.

Logic: added a temporary Loading view (a circle turning) waiting for credential checking result, once received, it will rerender whether `login` or the actual `path`.